src/utils/utils.jl

```diff
@@ -535,7 +515,7 @@ end
 
 # TODO: old code here does not properly handle data with different numbers of points
```

Comment:
What about these TODOs

Reply:
Addressed (won't fix the handling data with different numbers of points but added a validation check for it)